core/api.go

```diff
@@ -316,6 +317,22 @@ func blockchainIDHandler(handler http.Handler, blockchainID string) http.Handler
 	})
 }
 
+// loggingHandler pulls out request data and adds it to the request's
+// logging context.
+func loggingHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		ctx = log.AddPrefixkv(ctx, "path", req.URL.Path)
+		if userAgent := req.UserAgent(); userAgent != "" {
+			ctx = log.AddPrefixkv(ctx, "useragent", userAgent)
```

Comment:
If we want to avoid adding to the request context's logging prefix, another alternative is to propagate the request to httperror.Formatter.Write directly. It would be a bit invasive though because there are quite a few call sites, and I don't think all of them currently have access to the request.